File: tools/parity/dispatch/pi-overlay-ui-settings.ts
// TS dispatch for the pi-overlay-ui-settings parity module: maps the shared
// vector function names to the real `src/shared/pi-overlay-ui-settings.ts`
// exports so the harness compares the live TS reference against the Rust port.

import { mergePiOverlayUiSettings } from '../../../src/shared/pi-overlay-ui-settings'

export function dispatch(fn: string, input: unknown): unknown {
  switch (fn) {
    case 'mergePiOverlayUiSettings':
      return mergePiOverlayUiSettings(input)
    default:
      throw new Error(`unknown function ${fn}`)
  }
}

File: rust/crates/orca-parity/src/modules/pi_overlay_ui_settings.rs
//! Parity dispatch for `orca_config::pi_overlay_ui_settings` vs
//! `src/shared/pi-overlay-ui-settings.ts`.

use orca_config::merge_pi_overlay_ui_settings;
use serde_json::{json, Value};

pub fn dispatch(function: &str, input: &Value) -> Value {
    match function {
        // Returns the merged settings object directly; the port already shapes
        // it as a `serde_json::Value` matching `JSON.stringify` of the TS return.
        "mergePiOverlayUiSettings" => merge_pi_overlay_ui_settings(input),
        other => json!({ "__parity_error__": format!("unknown function {other}") }),
    }
}
